File: chapter6/tf_tutorial/src/tf_broadcaster.cpp
#include <ros/ros.h>
#include <tf/LinearMath/Quaternion.h>
#include <tf/LinearMath/Vector3.h>
#include <tf/transform_broadcaster.h>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "tf_broadcaster");

  ros::NodeHandle n;

  tf::TransformBroadcaster tb;
  tf::StampedTransform st;

  ros::Rate r(1.0);

  while(ros::ok())
  {
    tf::Vector3 translation;
    translation.setValue(0, 0, 1.0);
    st.setOrigin(translation);

    tf::Quaternion rotation;
    rotation.setRPY(0, 0, 0);
    st.setRotation(rotation);

    tb.sendTransform(tf::StampedTransform(st, ros::Time::now(), "frame1", "frame2"));
    ROS_INFO("Transform Published");

    r.sleep();
  }

  return 0;
}

File: chapter6/tf_tutorial/scripts/tf_broadcaster.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import rospy
import tf

if __name__ == '__main__':
    rospy.init_node('tf_broadcaster')

    br = tf.TransformBroadcaster()

    r = rospy.Rate(1.0)
    
    while not rospy.is_shutdown():
        translation = (0, 0, 1.0)
        rotation = tf.transformations.quaternion_from_euler(0,0,0, axes='sxyz')
        br.sendTransform(translation, rotation,
                         rospy.Time.now(),'frame2', 'frame1')
        rospy.loginfo('Transform Published')
        r.sleep()
